src/python/pants/help/help_info_extracter.py

```diff
@@ -1106,9 +1109,16 @@ def get_option_help_info(self, args, kwargs):
         removal_hint = kwargs.get("removal_hint")
         choices = self.compute_choices(kwargs)
 
-        dest = Parser.parse_dest(*args, **kwargs)
+        dest = parse_dest(*args, **kwargs)
         # Global options have three env var variants. The last one is the most human-friendly.
-        env_var = Parser.get_env_var_names(self._scope, dest)[-1]
+        udest = dest.upper()
+        if self._scope == GLOBAL_SCOPE:
```

Comment:
This is (simplified) logic moved from parser.py.

Reply:
for my information: we take only the last element, so we can discard the logic that computes the other values